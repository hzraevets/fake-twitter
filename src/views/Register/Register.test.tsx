import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
// @ts-ignore
import { create } from 'react-test-renderer';

import { Register } from './';

describe('Register', () => {
  it('render Register page', () => {
    render(<Register/>, { wrapper: BrowserRouter });

    // Page should show the title
    expect(screen.getByText('Create New Account')).toBeInTheDocument();
    // and the Register button
    expect(screen.getByText('Register')).toBeInTheDocument();
    // and the link to jump back to login page
    expect(screen.getByText('Already have an account?')).toBeInTheDocument();
    // check <input/> element in the page
    expect(screen.getByPlaceholderText('username')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('first name')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('password')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('re-enter password')).toBeInTheDocument();
  });

  it('render Register to snapshot', () => {
    const renderer = create(
      <BrowserRouter>
        <Register/>
      </BrowserRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
