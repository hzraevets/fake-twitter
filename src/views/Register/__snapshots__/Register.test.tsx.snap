// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register render Register to snapshot 1`] = `
<div
  className="register-container h-full flex"
>
  <div
    className="register m-auto"
  >
    <div
      className="ant-card ant-card-bordered h-full css-dev-only-do-not-override-diro6f"
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="absolute top-2 right-2 cursor-pointer select-none"
          onClick={[Function]}
        >
          <button
            aria-checked={false}
            className="ant-switch css-dev-only-do-not-override-diro6f"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <span
                className="ant-switch-inner-checked"
              />
              <span
                className="ant-switch-inner-unchecked"
              />
            </span>
          </button>
        </div>
        <h1
          className="text-2xl font-bold mt-0 mb-5"
        >
          Create New Account
        </h1>
        <form
          onSubmit={[Function]}
        >
          <div
            className="ant-space css-dev-only-do-not-override-diro6f ant-space-vertical w-72 md:w-96"
            style={Object {}}
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginBottom": 24,
                }
              }
            >
              <div
                className="ant-space css-dev-only-do-not-override-diro6f ant-space-vertical w-full"
                style={Object {}}
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginBottom": 16,
                    }
                  }
                >
                  <input
                    className="ant-input css-dev-only-do-not-override-diro6f"
                    disabled={false}
                    name="username"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="username"
                    style={Object {}}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginBottom": 16,
                    }
                  }
                >
                  <input
                    className="ant-input css-dev-only-do-not-override-diro6f"
                    disabled={false}
                    name="firstname"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="first name"
                    style={Object {}}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginBottom": 16,
                    }
                  }
                >
                  <input
                    className="ant-input css-dev-only-do-not-override-diro6f"
                    disabled={false}
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="password"
                    style={Object {}}
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <input
                    className="ant-input css-dev-only-do-not-override-diro6f"
                    disabled={false}
                    name="rePassword"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="re-enter password"
                    style={Object {}}
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <div
                className="ant-space css-dev-only-do-not-override-diro6f ant-space-horizontal ant-space-align-center float-right"
                style={Object {}}
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginRight": 24,
                    }
                  }
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                  >
                    Already have an account?
                  </a>
                </div>
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <button
                    className="ant-btn css-dev-only-do-not-override-diro6f ant-btn-primary"
                    disabled={false}
                    onClick={[Function]}
                    type="submit"
                  >
                    <span>
                      Register
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
